Vendored Frotz code makes Jericho hard to build

It seems there are compilation issues with newer GCC versions.

The following builds fine, for Debian 12:

```
docker run -it --rm  --user 1000  -v "$(pwd)":"$(pwd)"   -w "$(pwd)" python:3.11-bookworm bash -c "export HOME=$(pwd); pip install -e ."
```

But if you use Debian 13:

```
docker run -it --rm  --user 1000  -v "$(pwd)":"$(pwd)"   -w "$(pwd)" python:3.11-trixie bash -c "export HOME=$(pwd); pip install -e ."
```

you get the following

```
      src/ztools/showdict.c:37:58: warning: unused variable ‘length’ [-Wunused-variable]
         37 |     unsigned int separator_count, word_size, word_count, length;
            |                                                          ^~~~~~
      make: *** [Makefile:304: src/ztools/showdict.o] Error 1
      make: *** Waiting for unfinished jobs....
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/home/art/src/jericho/setup.py", line 25, in <module>
          subprocess.check_call(['make', 'library', '-j', '4'], cwd=FROTZPATH)
        File "/usr/local/lib/python3.11/subprocess.py", line 413, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['make', 'library', '-j', '4']' returned non-zero exit status 2.
```

I investigated updating the Frotz code in `frotz` but it does not appear to map exactly to any released Frotz version (closest I found was [this commit](https://gitlab.com/DavidGriffith/frotz/-/commits/caf63b49c4bf9d46632fde40937892b0cae6441e)) and the original commit in this jericho repo had additional files such as `src/games`.

--- a/frotz/src/games/anchor.c
+++ b/frotz/src/games/anchor.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -24,9 +24,9 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // Anchorhead: http://ifdb.tads.org/viewgame?id=op0uw1gn1tjqmjt7
 
-const char *anchor_intro[] = { "\n", "\n", "\n" };
+char *anchor_intro[] = { "\n", "\n", "\n" };
 
-const zword anchor_special_ram_addrs[4] = {
+zword anchor_special_ram_addrs[4] = {
   21922, // Combination lock
   40660, // Bathe
   38470, // Transitions between days

--- a/frotz/src/games/curses.c
+++ b/frotz/src/games/curses.c
@@ -36,7 +36,7 @@ const zword curses_special_ram_addrs[9] = {
   20911, // Turn sceptre
 };
 
-const char *curses_intro[] = { "\n" };
+char *curses_intro[] = { "\n" };
 
 zword* curses_ram_addrs(int *n) {
     *n = 9;

--- a/frotz/src/games/hollywood.c
+++ b/frotz/src/games/hollywood.c
@@ -32,9 +32,11 @@ const zword hollywood_special_ram_addrs[5] = {
   8381, // Safe dial
 };
 
-const char *hollywood_intro[] = { "turn statue west\n",
-                                  "turn statue east\n",
-                                  "turn statue north\n" };
+char *hollywood_intro[] = {
+  "turn statue west\n",
+  "turn statue east\n",
+  "turn statue north\n"
+};
 
 zword* hollywood_ram_addrs(int *n) {
     *n = 5;

--- a/frotz/src/games/huntdark.c
+++ b/frotz/src/games/huntdark.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -31,7 +31,7 @@ const zword huntdark_special_ram_addrs[4] = {
   8824, // Bats
 };
 
-const char *huntdark_intro[] = { "\n", "\n" };
+char *huntdark_intro[] = { "\n", "\n" };
 
 zword* huntdark_ram_addrs(int *n) {
     *n = 4;

--- a/frotz/src/games/jewel.c
+++ b/frotz/src/games/jewel.c
@@ -24,7 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // The Jewel of Knowledge: http://ifdb.tads.org/viewgame?id=hu60gp1bgkhlo5yx
 
-const char *jewel_intro[] = { "bypass\n", "yes\n" };
+char *jewel_intro[] = { "bypass\n", "yes\n" };
 
 zword* jewel_ram_addrs(int *n) {
     *n = 0;

--- a/frotz/src/games/library.c
+++ b/frotz/src/games/library.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -24,7 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // All Quiet on the Library Front: http://ifdb.tads.org/viewgame?id=400zakqderzjnu1i
 
-const char *library_intro[] = { "\n" };
+char *library_intro[] = { "\n" };
 
 zword* library_ram_addrs(int *n) {
     *n = 0;

--- a/frotz/src/games/loose.c
+++ b/frotz/src/games/loose.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -24,7 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // Mother Loose: http://ifdb.tads.org/viewgame?id=4wd3lyaxi4thp8qi
 
-const char *loose_intro[] = { "\n" };
+char *loose_intro[] = { "\n" };
 
 zword* loose_ram_addrs(int *n) {
     *n = 0;

--- a/frotz/src/games/lurking.c
+++ b/frotz/src/games/lurking.c
@@ -34,10 +34,12 @@ const zword lurking_special_ram_addrs[6] = {
   883, // Cutting line
 };
 
-const char *lurking_intro[] = { "sit on chair\n",
-                                "turn pc on\n",
-                                "login 872325412\n",
-                                "password uhlersoth\n" };
+char *lurking_intro[] = {
+  "sit on chair\n",
+  "turn pc on\n",
+  "login 872325412\n",
+  "password uhlersoth\n"
+};
 
 zword* lurking_ram_addrs(int *n) {
     *n = 6;

--- a/frotz/src/games/partyfoul.c
+++ b/frotz/src/games/partyfoul.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -24,8 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // Party Foul - http://ifdb.tads.org/viewgame?id=cqwq699i9qiqdju
 
-const char *partyfoul_intro[] = { "\n",
-                                  "no\n" };
+char *partyfoul_intro[] = { "\n", "no\n" };
 
 zword* partyfoul_ram_addrs(int *n) {
     *n = 0;

--- a/frotz/src/games/seastalker.c
+++ b/frotz/src/games/seastalker.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -24,7 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // Seastalker: http://ifdb.tads.org/viewgame?id=56wb8hflec2isvzm
 
-const char *seastalker_intro[] = { "John\n", "Wayne\n", "yes\n" };
+char *seastalker_intro[] = { "John\n", "Wayne\n", "yes\n" };
 
 const zword seastalker_special_ram_addrs[1] = {
   9740, // Tracks scimitar location
@@ -32,7 +32,7 @@ const zword seastalker_special_ram_addrs[1] = {
 
 zword* seastalker_ram_addrs(int *n) {
     *n = 1;
-    return seastalker_ram_addrs;
+    return seastalker_special_ram_addrs;
 }
 
 char** seastalker_intro_actions(int *n) {

--- a/frotz/src/games/sherbet.c
+++ b/frotz/src/games/sherbet.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -24,7 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // The Meteor, the Stone and a Long Glass of Sherbet: http://ifdb.tads.org/viewgame?id=273o81yvg64m4pkz
 
-const char *sherbet_intro[] = { " \n" };
+char *sherbet_intro[] = { " \n" };
 
 zword* sherbet_ram_addrs(int *n) {
     *n = 0;

--- a/frotz/src/games/sherlock.c
+++ b/frotz/src/games/sherlock.c
@@ -24,7 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // Sherlock: http://ifdb.tads.org/viewgame?id=ug3qu521hze8bsvz
 
-const char *sherlock_intro[] = { "\n" };
+char *sherlock_intro[] = { "\n" };
 
 zword* sherlock_ram_addrs(int *n) {
     *n = 0;

--- a/frotz/src/games/snacktime.c
+++ b/frotz/src/games/snacktime.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -24,7 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // Snack Time!: http://ifdb.tads.org/viewgame?id=yr3y8s9k8e40hl5q
 
-const char *snacktime_intro[] = { "\n" };
+char *snacktime_intro[] = { "\n" };
 
 const zword snacktime_special_ram_addrs[2] = {
   20743, // Used to quantify how awake the pet is

--- a/frotz/src/games/theatre.c
+++ b/frotz/src/games/theatre.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -29,7 +29,7 @@ const zword theatre_special_ram_addrs[2] = {
   17834, // Track watch hand
 };
 
-const char *theatre_intro[] = { "\n" };
+char *theatre_intro[] = { "\n" };
 
 zword* theatre_ram_addrs(int *n) {
     *n = 2;

--- a/frotz/src/games/trinity.c
+++ b/frotz/src/games/trinity.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -24,7 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // Trinity: http://ifdb.tads.org/viewgame?id=j18kjz80hxjtyayw
 
-const char *trinity_intro[] = { "\n" };
+char *trinity_intro[] = { "\n" };
 
 zword* trinity_ram_addrs(int *n) {
     *n = 0;

--- a/frotz/src/games/tryst.c
+++ b/frotz/src/games/tryst.c
@@ -1,4 +1,4 @@
-/* 
+/*
 Copyright (C) 2018 Microsoft Corporation
 
 This program is free software; you can redistribute it and/or
@@ -45,7 +45,7 @@ const zword tryst_special_ram_addrs[18] = {
   9362, // Door
 };
 
-const char *tryst_intro[] = { "\n" };
+char *tryst_intro[] = { "\n" };
 
 zword* tryst_ram_addrs(int *n) {
     *n = 18;

--- a/frotz/src/interface/frotz_interface.c
+++ b/frotz/src/interface/frotz_interface.c
@@ -1649,7 +1649,7 @@ void get_object(zobject *obj, zword obj_num) {
   }
 
   (*obj).num = obj_num;
-  get_text(0, obj_name_addr+1, &(*obj).name);
+  get_text(0, obj_name_addr+1, (*obj).name);
 
   (*obj).parent = get_parent(obj_num);
   (*obj).sibling = get_sibling(obj_num);

--- a/frotz/src/ztools/showdict.c
+++ b/frotz/src/ztools/showdict.c
@@ -86,7 +86,7 @@ int dict_size;
 
         /* Display the text for the word */
 
-        decode_text_to_buffer(&word_address, &(dict_word->word));
+        decode_text_to_buffer(&word_address, dict_word->word);
 
         /* Get the data for each entry */
 

